src/renderers/dom/server/ReactServerRenderingTransaction.js

```diff
@@ -55,6 +58,52 @@ var Mixin = {
   },
 
   /**
+   * @return {object} The queue to collect React async events.
+   */
+  getUpdateQueue: function() {
+    // We use a ReactNoopUpdateQueue that switches to ReactUpdateQueue while
+    // mounting
+    var transaction = this;
+    return {
```

Comment:
This allocates an object any time `getUpdateQueue` is called. Instead, it should only be initialized once. For example, you could write a class called `ReactServerUpdateQueue` that would take `transaction` as a constructor argument.